server/plugin/authorization.go

```diff
@@ -5,26 +5,20 @@ package plugin
 
 import (
 	"strings"
+
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils/bot"
 )
 
 // isAuthorized returns true if the user is authorized to use the workflow plugin's admin-level APIs/commands.
-func (p *Plugin) IsAuthorizedAdmin(mattermostID string) (bool, error) {
-	user, err := p.API.GetUser(mattermostID)
+func (p *Plugin) IsAuthorizedAdmin(mattermostUserID string) (bool, error) {
+	user, err := p.API.GetUser(mattermostUserID)
 	if err != nil {
 		return false, err
 	}
-	if strings.Contains(user.Roles, "system_admin") || p.userAuthorized(mattermostID) {
+	if strings.Contains(user.Roles, "system_admin") {
 		return true, nil
 	}
-	return false, nil
-}
-
-func (p *Plugin) userAuthorized(mattermostID string) bool {
-	list := strings.Split(p.getConfig().AdminUserIDs, ",")
-	for _, u := range list {
-		if mattermostID == strings.TrimSpace(u) {
-			return true
-		}
-	}
-	return false
+	conf := p.getConfig()
+	bot := bot.NewBot(p.API, conf.BotUserID).WithConfig(conf.BotConfig)
+	return bot.IsUserAdmin(mattermostUserID), nil
```

Comment:
Sorry, I'm confused about this... Why would IsUserAdmin belong in the bot namespace? We're creating a newbot just to call a utility function, which deals with authorization -- shouldn't it just be here like before?

Reply:
Well, the idea is that these users are "bot admins". As such, they may receive plugin's log messages, with this PR. They also get a (simple) check for being able to administer other plugin functions.

Reply:
Also, maybe the names `bot` and `NewBot` are partly to blame for the confusion here, not intention-revealing enough. `BotAPI`?

Reply:
`BotAPI`, or `GetBot` maybe? `NewBot` definitely through me for a second.

Reply:
I like `GetBot` - doesn't require another "API", yet reveals the light-weight nature of the call, hopefully. I also really like how `bot.DM(...)` looks, would like to keep `bot`.

Reply:
@cpoile now better?

Reply:
👍